fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -206,41 +216,56 @@ def create_report2(
             df_kw = df_kw.sort_values(['timestamp'])  # oldest first
             threads: List[str] = list(df_kw['subject'].unique())
             # Average thread length
+            # TODO: Refactor to pandas?
             tot_thread_len = 0
-            for thread in range(len(threads)):
-                df_thread = df_kw[df_kw['subject'] == threads[thread]]
+            thread_data: Dict[str, pd.DataFrame] = {}
```

Comment:
@rohaher I think I see where you might have been asking me to improve performance. I think this should improve speed a lot. Only need to subset thread dataframe once as opposed to 3x.